service-front/module/Application/src/Model/Service/Lpa/Communication.php

```diff
@@ -92,6 +86,114 @@ public function sendRegistrationCompleteEmail(Lpa $lpa)
         return true;
     }
 
+    public function setUpEmailFieldsForOnlinePayment(Lpa $lpa)
+    {
+        $this->emailTemplateRef = AbstractEmailService::EMAIL_LPA_REGISTRATION_WITH_PAYMENT1;
+
+        $amount = '';
+        if (isset($lpa->payment->amount)) {
+            $amount = $this->moneyFormat($lpa->payment->amount);
+        }
+
+        // Assume datetimes are in Europe/London timezone as all our users are in the UK
+        $paymentDate = '';
+        $refundDate = '';
+        if (isset($lpa->payment->date)) {
+            $lpa->payment->date->setTimezone(new DateTimeZone('Europe/London'));
+            $paymentDate = $lpa->payment->date->format('j F Y - g:ia');
+            $refundDate = $lpa->payment->date->add(new DateInterval('P42D'))->format('j F Y');
+        }
+
+        $this->data = array_merge($this->data, [
+            'lpaTypeTitleCase' => $this->lpaTypeTitleCase,
+            'lpaPaymentReference' => $lpa->payment->reference,
+            'lpaPaymentDate' => $paymentDate,
+            'paymentAmount' => $amount,
+            'date' => $refundDate,
+        ]);
+
+        // If we have a separate payment address, send the email to that also
+        if (!empty($lpa->payment->email) && ((string)$lpa->payment->email != strtolower($this->userEmailAddress))) {
+            $this->to[] = (string) $lpa->payment->email;
+        }
+    } 
+
+    public function setUpEmailFieldsForChequePayment(Lpa $lpa)
+    {
+        $this->emailTemplateRef = AbstractEmailService::EMAIL_LPA_REGISTRATION_WITH_CHEQUE_PAYMENT2;
+
+        $amount = '';
+        if (isset($lpa->payment->amount)) {
+            $amount = $this->moneyFormat($lpa->payment->amount);
+        }
+
+        $this->data = array_merge($this->data, [
+            'feeAmount' => $amount,
+        ]);
+    }
+
+    public function setUpEmailFieldsForNoPayment(Lpa $lpa)
+    {
+        $this->emailTemplateRef = AbstractEmailService::EMAIL_LPA_REGISTRATION_WITH_NO_PAYMENT3;
+
+        if (empty($lpa->document->peopleToNotify)) { 
+                $this->data = array_merge($this->data, [
+                    'PTN' => false,
+                ]);
+        }
+        else {
+                $this->data = array_merge($this->data, [
+                    'PTN' => true,
+                ]);
+        }
+    }
+
+    public function setUpEmailFieldsForPayments(Lpa $lpa)
+    {
+        // fill out email fields appropriately that apply to cheque and online payments
+        //
+        if (!empty($lpa->document->peopleToNotify)) { 
```

Comment:
I'd probably switch these clauses round so you have empty() first then !empty() (implied), rather than !empty() then (implied) empty().

Reply:
+1. Done :)